### ISSUE TITLE
Unexpected behavior for setting up build env

### ISSUE THREAD

**Issue by ijstokes** [Anaconda-Server/anaconda-build#142]
Two observations:

1. It is weird that the build process creates a conda environment in the directory `/opt/anaconda/env`.  That is the standard **container** directory which holds all environments, such as `/opt/anaconda/env/foo`.  No one would ever expect to `source activate /opt/anaconda/env`, but instead it should be `/opt/anaconda/env/foo`.  While this is somewhat of a technicality, it is weird that this was done and a little concerning that it is indicative of a lack of understanding of standard conda utilization.

2. There appears to be some kind of initial "setup" that is, in fact, removing very standard packages that then immediately get re-downloaded and installed.  Again, how weird!  Any actions like that should, IMO, happen in the process of creating the base build container image (i.e. once period), not at the start of every build.  Things like `python` (!!!), `setuptools`, `pip`, `readline` get removed and then immediately downloaded again and re-added.  Wacky!

Examples:

* https://anaconda.org/nbcio/nbpresent/builds/1/0
* https://anaconda.org/ijstokes/serpent/builds/5/5

Footnote: it would be great to suppress the progress bars from the output, if that is possible.  I think the reset is valuable (and probably even necessary).

**Comment by ijstokes** [comment]
Another outcome of point 2: this leads to a whole awful lot of extraneous information at the start of the build log that has nothing to do with my package build.  Case in point:

* https://anaconda.org/ijstokes/nbpresent/builds/1/1

**Comment by srossross** [comment]
@ijstokes do you think this could move to flowdoc or a wiki?

**Comment by ijstokes** [comment]
@srossross I don't think this should be in Flowdock or a wiki: I think someone on the Anaconda Build team should consider my observations and either say "Actually, that is exactly how it is supposed to be for these good reasons" or "Huh, I agree, we should do something about those points, and here are three new issues to address the observations".
